Failure conda-forge Pythran on Github Actions Windows (because it uses MSVC)

It might be more a question than a real Pythran issue.

On GitHub Actions Windows, Pythran 0.18.0 fails with (see https://github.com/fluiddyn/transonic/actions/runs/17895355653/job/50881265988, related Merge Request https://foss.heptapod.net/fluiddyn/transonic/-/merge_requests/149).

```
Run pixi run pythran .github/minimal_pythran.py -vv
  pixi run pythran .github/minimal_pythran.py -vv
  shell: C:\Program Files\Git\bin\bash.EXE -l {0}
INFO          sys file exists: D:\a\transonic\transonic\.pixi\envs\default\Lib\site-packages\pythran\pythran.cfg
INFO     platform file exists: D:\a\transonic\transonic\.pixi\envs\default\Lib\site-packages\pythran\pythran-win32.cfg
WARNING      user file does not exist: C:\Users\runneradmin\.pythranrc
INFO     pythranrc section [pythran] is valid and options are correct
INFO     pythranrc section [typing] is valid and options are correct
INFO     pythranrc section [backend] is valid and options are correct
INFO     pythranrc section [compiler] is valid and options are correct
INFO     Pythran support for package 'omp' will be reduced: this module is not available at runtime.
No `packages` or `py_modules` configuration, performing automatic discovery.
`src-layout` detected -- analysing .\src
discovered packages -- ['transonic', 'transonic_cl', 'transonic.analyses', 'transonic.backends']
discovered py_modules -- []
running build_ext
building 'minimal_pythran' extension
creating C:\Users\RUNNER~1\AppData\Local\Temp\tmp8_q3rygf\Release\Users\RUNNER~1\AppData\Local\Temp
"C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.44.35207\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -DENABLE_PYTHON_MODULE -DPYTHRAN_BLAS_NONE -ID:/a/transonic/transonic/.pixi/envs/default/Library/include -ID:\a\transonic\transonic\.pixi\envs\default\Lib\site-packages/pythran -ID:\a\transonic\transonic\.pixi\envs\default\Lib\site-packages\numpy\_core\include -ID:\a\transonic\transonic\.pixi\envs\default\include -ID:\a\transonic\transonic\.pixi\envs\default\Include "-IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.44.35207\include" "-IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.44.35207\ATLMFC\include" "-IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\VS\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.26100.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.26100.0\\um" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.26100.0\\shared" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.26100.0\\winrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.26100.0\\cppwinrt" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um" /EHsc /TpC:\Users\RUNNER~1\AppData\Local\Temp\tmpmesa0prr.cpp /FoC:\Users\RUNNER~1\AppData\Local\Temp\tmp8_q3rygf\Release\Users\RUNNER~1\AppData\Local\Temp\tmpmesa0prr.obj -O2 /std:c++14 /w
cl : Command line warning D9025 : overriding '/W3' with '/w'
tmpmesa0prr.cpp
D:\a\transonic\transonic\.pixi\envs\default\Lib\site-packages/pythran\pythonic/include/types/ndarray.hpp(484): error C2559: 'std::enable_if<`anonymous-namespace'::pythonic::types::is_numexpr_arg<F>::value&&std::is_same<bool,F::dtype>::value&&F::value==1&&!`anonymous-namespace'::pythonic::types::is_pod_array<F>::value,`anonymous-namespace'::pythonic::types::numpy_vexpr<`anonymous-namespace'::pythonic::types::ndarray<T,pS>,`anonymous-namespace'::pythonic::types::ndarray<long,`anonymous-namespace'::pythonic::types::pshape<long>>>>::type `anonymous-namespace'::pythonic::types::ndarray<T,pS>::operator [](const F &) const': cannot overload a member function without ref-qualifier with a member function with ref-qualifier
[...]
```

I don't understand why Pythran uses "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.44.35207\bin\HostX86\x64\cl.exe".

I checked and neither `CC` or `CXX` are defined. Then, my understanding is that Pythran should use what is defined in 
`pythran-win32.cfg`, i.e. `clang-cl.exe`.

I guess I'm doing something wrong, but I don't see what.

The environment used has `clang` and co:

```
pixi list --locked
  C:\Users\runneradmin\.pixi\bin\pixi.exe list --locked --manifest-path pixi.toml --color always
  Package                    Version             Build                  Size       Kind   Source
  _python_abi3_support       1.0                 hd8ed1ab_2             8 KiB      conda  https://conda.anaconda.org/conda-forge/
  asttokens                  3.0.0               pyhd8ed1ab_1           27.5 KiB   conda  https://conda.anaconda.org/conda-forge/
  attrs                      25.3.0              pyh71513ae_0           55.8 KiB   conda  https://conda.anaconda.org/conda-forge/
  autopep8                   2.3.2               pyhd8ed1ab_0           45.1 KiB   conda  https://conda.anaconda.org/conda-forge/
  beniget                    0.4.2.post1         pyhd8ed1ab_1           21.2 KiB   conda  https://conda.anaconda.org/conda-forge/
  black                      25.1.0              pyh866005b_0           168.6 KiB  conda  https://conda.anaconda.org/conda-forge/
  bzip2                      1.0.8               h0ad9c76_8             54.7 KiB   conda  https://conda.anaconda.org/conda-forge/
  ca-certificates            2025.8.3            h4c7d964_0             150.9 KiB  conda  https://conda.anaconda.org/conda-forge/
  clang                      21.1.1              default_hac490eb_0     105.5 MiB  conda  https://conda.anaconda.org/conda-forge/
  clang-21                   21.1.1              default_hac490eb_0     35.9 MiB   conda  https://conda.anaconda.org/conda-forge/
  clangxx                    21.1.1              default_hac490eb_0     70.3 MiB   conda  https://conda.anaconda.org/conda-forge/
  click                      8.3.0               pyh7428d3b_0           90 KiB     conda  https://conda.anaconda.org/conda-forge/
  colorama                   0.4.6               pyhd8ed1ab_1           26.4 KiB   conda  https://conda.anaconda.org/conda-forge/
  colorlog                   6.9.0               pyh7428d3b_1           15.9 KiB   conda  https://conda.anaconda.org/conda-forge/
  comm                       0.2.3               pyhe01879c_0           14.3 KiB   conda  https://conda.anaconda.org/conda-forge/
  coverage                   7.10.6              py313hd650c13_1        403.2 KiB  conda  https://conda.anaconda.org/conda-forge/
  cpython                    3.13.7              py313hd8ed1ab_100      47 KiB     conda  https://conda.anaconda.org/conda-forge/
  cython                     3.1.4               py313hb1c8229_2        3.1 MiB    conda  https://conda.anaconda.org/conda-forge/
  debugpy                    1.8.17              py313h927ade5_0        3.8 MiB    conda  https://conda.anaconda.org/conda-forge/
  decorator                  5.2.1               pyhd8ed1ab_0           13.8 KiB   conda  https://conda.anaconda.org/conda-forge/
  distlib                    0.4.0                                      1.1 MiB    pypi   distlib-0.4.0-py2.py3-none-any.whl
  exceptiongroup             1.3.0               pyhd8ed1ab_0           20.8 KiB   conda  https://conda.anaconda.org/conda-forge/
  executing                  2.2.1               pyhd8ed1ab_0           30 KiB     conda  https://conda.anaconda.org/conda-forge/
  filelock                   3.19.1                                     41.5 KiB   pypi   filelock-3.19.1-py3-none-any.whl
  gast                       0.6.0               pyhd8ed1ab_0           24.3 KiB   conda  https://conda.anaconda.org/conda-forge/
  importlib-metadata         8.7.0               pyhe01879c_1           33.8 KiB   conda  https://conda.anaconda.org/conda-forge/
  iniconfig                  2.0.0               pyhd8ed1ab_1           11.2 KiB   conda  https://conda.anaconda.org/conda-forge/
  ipykernel                  6.30.1              pyh3521513_0           119.1 KiB  conda  https://conda.anaconda.org/conda-forge/
  ipython                    9.5.0               pyh6be1c34_0           615.4 KiB  conda  https://conda.anaconda.org/conda-forge/
  ipython_pygments_lexers    1.1.1               pyhd8ed1ab_0           13.7 KiB   conda  https://conda.anaconda.org/conda-forge/
  jedi                       0.19.2              pyhd8ed1ab_1           823.9 KiB  conda  https://conda.anaconda.org/conda-forge/
  jsonschema                 4.25.1              pyhe01879c_0           79.8 KiB   conda  https://conda.anaconda.org/conda-forge/
  jsonschema-specifications  2025.9.1            pyhcf101f3_0           18.8 KiB   conda  https://conda.anaconda.org/conda-forge/
  jupyter_client             8.6.3               pyhd8ed1ab_1           103.8 KiB  conda  https://conda.anaconda.org/conda-forge/
  jupyter_core               5.8.1               pyh5737063_0           58.6 KiB   conda  https://conda.anaconda.org/conda-forge/
  krb5                       1.21.3              hdf4eb48_0             695.3 KiB  conda  https://conda.anaconda.org/conda-forge/
  libblas                    3.9.0               35_h5709861_mkl        64.5 KiB   conda  https://conda.anaconda.org/conda-forge/
  libcblas                   3.9.0               35_h2a3cdd5_mkl        64.8 KiB   conda  https://conda.anaconda.org/conda-forge/
  libexpat                   2.7.1               hac47afa_0             138 KiB    conda  https://conda.anaconda.org/conda-forge/
  libffi                     3.4.6               h537db12_1             43.9 KiB   conda  https://conda.anaconda.org/conda-forge/
  libhwloc                   2.11.2              default_hc8275d1_1000  2.3 MiB    conda  https://conda.anaconda.org/conda-forge/
  libiconv                   1.18                hc1393d2_2             680.6 KiB  conda  https://conda.anaconda.org/conda-forge/
  liblapack                  3.9.0               35_hf9ab0e9_mkl        76.6 KiB   conda  https://conda.anaconda.org/conda-forge/
  liblzma                    5.8.1               h2466b09_2             102.5 KiB  conda  https://conda.anaconda.org/conda-forge/
  libmpdec                   4.0.0               h2466b09_0             86.6 KiB   conda  https://conda.anaconda.org/conda-forge/
  libsodium                  1.0.20              hc70643c_0             197.6 KiB  conda  https://conda.anaconda.org/conda-forge/
  libsqlite                  3.50.4              hf5d6505_0             1.2 MiB    conda  https://conda.anaconda.org/conda-forge/
  libxml2                    2.13.8              h741aa76_1             1.4 MiB    conda  https://conda.anaconda.org/conda-forge/
  libzlib                    1.3.1               h2466b09_2             54.2 KiB   conda  https://conda.anaconda.org/conda-forge/
  llvm-openmp                20.1.8              hfa2b4ca_2             336.4 KiB  conda  https://conda.anaconda.org/conda-forge/
  llvmlite                   0.44.0              py313hc403fe3_2        17.3 MiB   conda  https://conda.anaconda.org/conda-forge/
  m2w64-gcc-libgfortran      5.3.0               6                      342.5 KiB  conda  https://conda.anaconda.org/conda-forge/
  m2w64-gcc-libs             5.3.0               7                      519.9 KiB  conda  https://conda.anaconda.org/conda-forge/
  m2w64-gcc-libs-core        5.3.0               7                      214.1 KiB  conda  https://conda.anaconda.org/conda-forge/
  m2w64-gmp                  6.1.0               2                      726.1 KiB  conda  https://conda.anaconda.org/conda-forge/
  m2w64-libwinpthread-git    5.0.0.4634.697f757  2                      31.2 KiB   conda  https://conda.anaconda.org/conda-forge/
  matplotlib-inline          0.1.7               pyhd8ed1ab_1           14.1 KiB   conda  https://conda.anaconda.org/conda-forge/
  mkl                        2024.2.2            h57928b3_16            98.3 MiB   conda  https://conda.anaconda.org/conda-forge/
  mpi                        1.0                 msmpi                  4.6 KiB    conda  https://conda.anaconda.org/conda-forge/
  mpi4py                     4.1.0               py313he7c20cf_102      784.9 KiB  conda  https://conda.anaconda.org/conda-forge/
  msmpi                      10.1.1              h3502643_7             11.1 MiB   conda  https://conda.anaconda.org/conda-forge/
  msys2-conda-epoch          20160418            1                      3.2 KiB    conda  https://conda.anaconda.org/conda-forge/
  mypy_extensions            1.1.0               pyha770c72_0           11.5 KiB   conda  https://conda.anaconda.org/conda-forge/
  nbformat                   5.10.4              pyhd8ed1ab_1           98.6 KiB   conda  https://conda.anaconda.org/conda-forge/
  nbval                      0.11.0              pyhd8ed1ab_1           26 KiB     conda  https://conda.anaconda.org/conda-forge/
  nest-asyncio               1.6.0               pyhd8ed1ab_1           11.3 KiB   conda  https://conda.anaconda.org/conda-forge/
  numba                      0.61.2              py313h96c6e06_1        5.6 MiB    conda  https://conda.anaconda.org/conda-forge/
  numpy                      2.2.6               py313hefb8edb_0        6.8 MiB    conda  https://conda.anaconda.org/conda-forge/
  openssl                    3.5.3               h725018a_0             8.8 MiB    conda  https://conda.anaconda.org/conda-forge/
  packaging                  25.0                pyh29332c3_1           61 KiB     conda  https://conda.anaconda.org/conda-forge/
  parso                      0.8.5               pyhcf101f3_0           79.7 KiB   conda  https://conda.anaconda.org/conda-forge/
  pathspec                   0.12.1              pyhd8ed1ab_1           40.1 KiB   conda  https://conda.anaconda.org/conda-forge/
  pdm-backend                2.4.5               pyhd8ed1ab_0           180 KiB    conda  https://conda.anaconda.org/conda-forge/
  pickleshare                0.7.5               pyhd8ed1ab_1004        11.5 KiB   conda  https://conda.anaconda.org/conda-forge/
  pip                        25.2                pyh145f28c_0           1.1 MiB    conda  https://conda.anaconda.org/conda-forge/
  platformdirs               4.4.0               pyhcf101f3_0           23.1 KiB   conda  https://conda.anaconda.org/conda-forge/
  pluggy                     1.6.0               pyhd8ed1ab_0           23.7 KiB   conda  https://conda.anaconda.org/conda-forge/
  ply                        3.11                pyhd8ed1ab_3           47.9 KiB   conda  https://conda.anaconda.org/conda-forge/
  prompt-toolkit             3.0.52              pyha770c72_0           267.5 KiB  conda  https://conda.anaconda.org/conda-forge/
  psutil                     7.1.0               py313h5ea7bf4_0        484.7 KiB  conda  https://conda.anaconda.org/conda-forge/
  pthreads-win32             2.9.1               h2466b09_4             259.6 KiB  conda  https://conda.anaconda.org/conda-forge/
  pure_eval                  0.2.3               pyhd8ed1ab_1           16.3 KiB   conda  https://conda.anaconda.org/conda-forge/
  pycodestyle                2.14.0              pyhd8ed1ab_0           34.4 KiB   conda  https://conda.anaconda.org/conda-forge/
  pygments                   2.19.2              pyhd8ed1ab_0           868.4 KiB  conda  https://conda.anaconda.org/conda-forge/
  pytest                     8.4.2               pyhd8ed1ab_0           270.2 KiB  conda  https://conda.anaconda.org/conda-forge/
  pytest-cov                 7.0.0               pyhcf101f3_1           28.3 KiB   conda  https://conda.anaconda.org/conda-forge/
  pytest_venv                0.3                                        7.6 KiB    pypi   pytest_venv-0.3-py3-none-any.whl
  python                     3.13.7              hdf00ec1_100_cp313     15.6 MiB   conda  https://conda.anaconda.org/conda-forge/
  python-dateutil            2.9.0.post0         pyhe01879c_2           227.8 KiB  conda  https://conda.anaconda.org/conda-forge/
  python-fastjsonschema      2.21.2              pyhe01879c_0           238.9 KiB  conda  https://conda.anaconda.org/conda-forge/
  python-gil                 3.13.7              h4df99d1_100           47 KiB     conda  https://conda.anaconda.org/conda-forge/
  python_abi                 3.13                8_cp313                6.8 KiB    conda  https://conda.anaconda.org/conda-forge/
  pythran                    0.18.0              pyh7428d3b_0           1.9 MiB    conda  https://conda.anaconda.org/conda-forge/
  pywin32                    311                 py313h40c08fc_1        6.4 MiB    conda  https://conda.anaconda.org/conda-forge/
  pyzmq                      27.1.0              py312hbb5da91_0        181.4 KiB  conda  https://conda.anaconda.org/conda-forge/
  referencing                0.36.2              pyh29332c3_0           50.5 KiB   conda  https://conda.anaconda.org/conda-forge/
  rpds-py                    0.27.1              py313hfbe8231_1        244.4 KiB  conda  https://conda.anaconda.org/conda-forge/
  scipy                      1.16.2              py313h62a08ca_0        14.6 MiB   conda  https://conda.anaconda.org/conda-forge/
  setuptools                 80.9.0              pyhff2d567_0           731.2 KiB  conda  https://conda.anaconda.org/conda-forge/
  six                        1.17.0              pyhe01879c_1           18 KiB     conda  https://conda.anaconda.org/conda-forge/
  stack_data                 0.6.3               pyhd8ed1ab_1           26.4 KiB   conda  https://conda.anaconda.org/conda-forge/
  tbb                        2021.13.0           h62715c5_1             147.9 KiB  conda  https://conda.anaconda.org/conda-forge/
  tk                         8.6.13              h2c6b04d_2             3.3 MiB    conda  https://conda.anaconda.org/conda-forge/
  tomli                      2.2.1               pyhe01879c_2           20.7 KiB   conda  https://conda.anaconda.org/conda-forge/
  tornado                    6.5.2               py313h5ea7bf4_1        856 KiB    conda  https://conda.anaconda.org/conda-forge/
  traitlets                  5.14.3              pyhd8ed1ab_1           107.5 KiB  conda  https://conda.anaconda.org/conda-forge/
  typing_extensions          4.15.0              pyhcf101f3_0           50.5 KiB   conda  https://conda.anaconda.org/conda-forge/
  tzdata                     2025b               h78e105d_0             120.1 KiB  conda  https://conda.anaconda.org/conda-forge/
  ucrt                       10.0.26100.0        h57928b3_0             678.4 KiB  conda  https://conda.anaconda.org/conda-forge/
  vc                         14.3                h41ae7f8_31            17.8 KiB   conda  https://conda.anaconda.org/conda-forge/
  vc14_runtime               14.44.35208         h818238b_31            666.4 KiB  conda  https://conda.anaconda.org/conda-forge/
  vcomp14                    14.44.35208         h818238b_31            111.3 KiB  conda  https://conda.anaconda.org/conda-forge/
  virtualenv                 20.34.0                                    6.1 MiB    pypi   virtualenv-20.34.0-py3-none-any.whl
  vs2015_runtime             14.44.35208         h38c0c73_31            17.8 KiB   conda  https://conda.anaconda.org/conda-forge/
  wcwidth                    0.2.13              pyhd8ed1ab_1           31.8 KiB   conda  https://conda.anaconda.org/conda-forge/
  zeromq                     4.3.5               h5bddc39_9             259 KiB    conda  https://conda.anaconda.org/conda-forge/
  zipp                       3.23.0              pyhd8ed1ab_0           22.4 KiB   conda  https://conda.anaconda.org/conda-forge/
  zstd                       1.5.7               hbeecb71_2             346.4 KiB  conda  https://conda.anaconda.org/conda-forge/
```

--- a/pythran/dist.py
+++ b/pythran/dist.py
@@ -89,6 +89,8 @@ class PythranBuildExtMixIn(object):
         if getattr(ext, 'cc', None) is not None:
             try:
                 import distutils._msvccompiler as msvc
+                if not hasattr(msvc, "_find_exe"):
+                    msvc = msvc.msvc
                 # install hook
                 find_exe = msvc._find_exe
 
@@ -98,7 +100,7 @@ class PythranBuildExtMixIn(object):
                     return find_exe(exe, *args, **kwargs)
 
                 msvc._find_exe = _find_exe
-            except (AttributeError, ImportError):
+            except (AttributeError, ImportError) as e:
                 pass
 
         # In general, distutils uses -Wstrict-prototypes, but this option

